### ISSUE TITLE
Session is undefined in calendar.js

### ISSUE THREAD

**Issue by nbourre** [microsoftgraph/msgraph-training-nodeexpressapp#66]
### Where did you get the code?

- [ ] Downloaded from GitHub
- [ ] Downloaded from the [Microsoft Graph quick start tool](https://developer.microsoft.com/graph/quick-start)
- [X] Followed the tutorial from [Microsoft Graph tutorials](https://docs.microsoft.com/graph/tutorials)

### Describe the bug
In the part [Get Calendar View](https://docs.microsoft.com/en-us/graph/tutorials/node?tutorial-step=4), in step 2, the line 11 `if (!req.session.userId) {`, I get the error "Cannot read property 'userId' of undefined".

### To Reproduce

1. Go to landing page
2. Sign in
3. Click on 'Calendar'

### Expected behavior

A json dump of the calendar events.

### Screenshots

![kFG8umCjLN](https://user-images.githubusercontent.com/2332679/117027153-6c94e280-acca-11eb-8c64-d9dc760f66a2.gif)

### Desktop

- OS: Windows
- Browser : Firefox
- Version : 88

### Other information
It might have something to do with the 3rd step in the page "Create a Node.js Express apps with web app" where there's a secret marked as 'your_secret_value_here'. But nowhere in the tutorial, there's a mention of what do to with this line.

**Comment by jasonjoh** [comment]
The `your_secret_value_here` is fine for a sample, it still works. It's meant to be some secret value for signing the sessions, but it can be any string. (See http://expressjs.com/en/resources/middleware/session.html#secret).

It sounds like your session is either not being created or is getting lost. One thing to check is compare your app.js with the one in this repo - Express middleware can be a little tricky if you don't declare things in a particular order.

Another thing to check is how you're browsing to the app locally. You must go to http://localhost:3000, NOT http://127.0.0.1:3000.

**Comment by nbourre** [comment]
I've copy-pasted the app.js from this repo. Unfortunately, I get the same error.

Can it be because I run the server in WSL2?

**Comment by jasonjoh** [comment]
I wouldn't expect WSL2 to cause a problem, but it sounds like something isn't right with the session middleware. `req.session` shouldn't be undefined, even if the app hasn't set anything in it. If you wrap this (in auth.js) in a try/catch do you get any errors?

```
// Save the user's homeAccountId in their session
req.session.userId = response.account.homeAccountId;
```

If you have your app in a GitHub repo I'm happy to take a look at it and see if I can spot the problem.

**Comment by jasonjoh** [comment]
Great! Good news - I reproduce the error. Let me see what I can find.

**Comment by nbourre** [comment]
I think it might be something with the "Install Node packages" part of the tutorial.

It's already been a few days, but I think there was some kind of warning at the moment I imported them.

**Comment by jasonjoh** [comment]
I noticed that you had this line near the top of app.js:

```
app.use('/calendar', calendarRouter);
```

This put it before the `app.use(session({` line, which means that the session middleware doesn't exist when you define you calendar route. The calendar route gets an undefined session instead, causing the error. :D.

I moved that line down just after `app.use('/auth', authRouter);` and the error went away.
